editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -3585,81 +3585,86 @@ void CanvasItemEditor::_update_scrollbars() {
-		if (ABS(begin.y - previous_update_view_offset.y) < ABS(begin.y - view_offset.y)) {
-			view_offset.y = previous_update_view_offset.y;
-		}
+		if (canvas_item_rect.size.height <= (local_rect.size.y / zoom)) {
+			if (ABS(begin.y - previous_update_view_offset.y) < ABS(begin.y - view_offset.y)) {
+				view_offset.y = previous_update_view_offset.y;
+			}
 
-		v_scroll->hide();
-	} else {
-		if (view_offset.y > end.y && view_offset.y > previous_update_view_offset.y) {
-			view_offset.y = MAX(end.y, previous_update_view_offset.y);
-		}
-		if (view_offset.y < begin.y && view_offset.y < previous_update_view_offset.y) {
-			view_offset.y = MIN(begin.y, previous_update_view_offset.y);
+			v_scroll->hide();
+		} else {
+			if (view_offset.y > end.y && view_offset.y > previous_update_view_offset.y) {
+				view_offset.y = MAX(end.y, previous_update_view_offset.y);
+			}
+			if (view_offset.y < begin.y && view_offset.y < previous_update_view_offset.y) {
+				view_offset.y = MIN(begin.y, previous_update_view_offset.y);
+			}
+
+			v_scroll->show();
+			v_scroll->set_min(MIN(view_offset.y, begin.y));
+			v_scroll->set_max(MAX(view_offset.y, end.y) + screen_rect.y);
+			v_scroll->set_page(screen_rect.y);
 		}
 
-		v_scroll->show();
-		v_scroll->set_min(MIN(view_offset.y, begin.y));
-		v_scroll->set_max(MAX(view_offset.y, end.y) + screen_rect.y);
-		v_scroll->set_page(screen_rect.y);
-	}
+		if (canvas_item_rect.size.width <= (local_rect.size.x / zoom)) {
+			if (ABS(begin.x - previous_update_view_offset.x) < ABS(begin.x - view_offset.x)) {
+				view_offset.x = previous_update_view_offset.x;
+			}
 
-	if (canvas_item_rect.size.width <= (local_rect.size.x / zoom)) {
-		if (ABS(begin.x - previous_update_view_offset.x) < ABS(begin.x - view_offset.x)) {
-			view_offset.x = previous_update_view_offset.x;
-		}
+			h_scroll->hide();
+		} else {
+			if (view_offset.x > end.x && view_offset.x > previous_update_view_offset.x) {
+				view_offset.x = MAX(end.x, previous_update_view_offset.x);
+			}
+			if (view_offset.x < begin.x && view_offset.x < previous_update_view_offset.x) {
+				view_offset.x = MIN(begin.x, previous_update_view_offset.x);
+			}
 
-		h_scroll->hide();
-	} else {
-		if (view_offset.x > end.x && view_offset.x > previous_update_view_offset.x) {
-			view_offset.x = MAX(end.x, previous_update_view_offset.x);
-		}
-		if (view_offset.x < begin.x && view_offset.x < previous_update_view_offset.x) {
-			view_offset.x = MIN(begin.x, previous_update_view_offset.x);
+			h_scroll->show();
+			h_scroll->set_min(MIN(view_offset.x, begin.x));
+			h_scroll->set_max(MAX(view_offset.x, end.x) + screen_rect.x);
+			h_scroll->set_page(screen_rect.x);
 		}
 
-		h_scroll->show();
-		h_scroll->set_min(MIN(view_offset.x, begin.x));
-		h_scroll->set_max(MAX(view_offset.x, end.x) + screen_rect.x);
-		h_scroll->set_page(screen_rect.x);
+		// Calculate scrollable area
+		v_scroll->set_value(view_offset.y);
+		h_scroll->set_value(view_offset.x);
+	} else {
+		h_scroll->hide();
```

Comment:
I don't think this in good to completely disable the scrollbars. That's the complex part of the problem.
Those bars should still be present if possible, but should adapt depending on where you are.

Reply:
Totally changed.